api/v1alpha1/volumereplicationgroup_types.go

```diff
@@ -18,53 +18,32 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	vrv1alpha1 "github.com/kube-storage/volume-replication-operator/api/v1alpha1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// VolumeTakeoverControlType -- takeover control when desiredCluster is different from affinedCluster
-type VolumeTakeoverControlType string
-
-// VolumeTakeoverControlType definitions
-const (
-	// Force promote the volume in a WAN DR setting
-	ForcePromote VolumeTakeoverControlType = "ForcePromote"
-)
-
-// VolumeReplicationGroupSpec defines the desired state of VolumeReplicationGroup
+// VolumeReplicationGroup (VRG) spec declares the desired replication class
+// and replication state of all the PVCs owned by a given application, which
+// are identified via the given application label selector.  The VRG operator
+// creates children VolumeReplication (VR) CRs for each PVC of the application,
+// with the desired replication class and replication image state (primary or
+// secondary).
 type VolumeReplicationGroupSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
 	// Name of the application to replicate
-	ApplicationName string `json:"applicationName"`
-
-	// Application label that is used to identify its PVs
-	ApplicationLabels metav1.LabelSelector `json:"applicationLabels"`
-
-	// Cluster ID in ClusterPeers that has best storage performance affinity for the application
-	AffinedCluster string `json:"affinedCluster"`
-
-	// Desired cluster that should takeover the application from current active cluster.
-	// May be set either to:
-	// - secondary cluster ID (to migrate application or takeover application in case of a disaster)
-	// - nil (if application is not active in the affined cluster, takeback to affined cluster)
-	// +optional
-	DesiredCluster string `json:"desiredCluster,omitempty"`
+	ApplicationName string `json:"applicationName,omitempty"`
```

Comment:
Do we need a name here?

Reply:
The intent behind the optional ApplicationName is to help the user correlate the VRG with the application, especially if there can be multiple applications running in the same namespace.  If VRG name is sufficient to correlate, then this field can be removed.

Reply:
This correlation requirement could also be solved by AVR adding a label to the VRG: {app = AVR_name}.
In addition, AVR doesn't readily have an application name.

Reply:
Created https://github.com/RamenDR/ramen/issues/14 to track this.